ouroboros-consensus/src/Ouroboros/Consensus/BlockchainTime.hs

```diff
@@ -106,38 +91,6 @@ blockUntilSlot btime slot = do
 onSlotChange :: HasCallStack => BlockchainTime m -> (SlotNo -> m ()) -> m ()
 onSlotChange = onSlotChange_
 
--- | Wrapper around 'onSlot_' to ensure 'HasCallStack' constraint
-onSlot :: HasCallStack => BlockchainTime m -> SlotNo -> m () -> m ()
-onSlot = onSlot_
-
--- | Default implementation of 'onSlot' (used internally only)
-defaultOnSlot :: forall m. (IOLike m, HasCallStack)
-              => ResourceRegistry m
-              -> STM m SlotNo
-              -> SlotNo
-              -> m ()
-              -> m ()
-defaultOnSlot registry getCurrentSlot slot action = do
-    startingSlot <- atomically getCurrentSlot
-    when (startingSlot >= slot) $
-      throwM $ OnSlotTooLate slot startingSlot
-    runWhenJust registry waitForSlot (\() -> action)
-  where
-    waitForSlot :: STM m (Maybe ())
-    waitForSlot = do
-        currentSlot <- getCurrentSlot
-        return $ if currentSlot >= slot
-                   then Just ()
-                   else Nothing
-
-data OnSlotException =
-    -- | An action was scheduled via 'onSlot' for a slot in the past.
-    -- First slot is requested, second slot is current as of raising.
-    OnSlotTooLate SlotNo SlotNo
-  deriving (Eq, Show)
-
-instance Exception OnSlotException
-
```

Comment:
@nfrisby I remember you added this. What do you think about this change?

Reply:
(If the exception is needed, we can re-introduce it in the test.)

Reply:
Rebased, and moved this to `Test.Util.BlockchainTime` where it is now defined as

```haskell
onSlot :: (HasCallStack, IOLike m) => BlockchainTime m -> SlotNo -> m () -> m ()
onSlot btime slot k = do
    startingSlot <- atomically $ getCurrentSlot btime
    when (startingSlot >= slot) $
      throwM $ OnSlotTooLate slot startingSlot
    onSlotChange btime $ \slot' ->
      when (slot == slot') k

data OnSlotException =
    -- | An action was scheduled via 'onSlot' for a slot in the past.
    -- First slot is requested, second slot is current as of raising.
    OnSlotTooLate SlotNo SlotNo
  deriving (Eq, Show)

instance Exception OnSlotException
```

Reply:
Moving it is no problem. Thanks for the notice.